lib/prawn/fonts/ttf.rb

```diff
@@ -364,14 +363,22 @@ def embed(reference, subset)
         cmap.stream.compress!
 
         reference.data.update(
-          Subtype: :TrueType,
           BaseFont: basename.to_sym,
           FontDescriptor: descriptor,
           FirstChar: 32,
           LastChar: 255,
           Widths: @document.ref!(widths),
           ToUnicode: cmap
         )
+
+        if font.cff.exists?
```

Comment:
Would it be useful/more readable to have this extracted into a separate method and have an OTF subclass that overrides it?

Reply:
On second thought, I think this is correct as-is. It appears Prawn looks at the file extension to determine which `Font` subclass to use. That's not _wrong_ necessarily, but it doesn't prevent fonts with a .ttf file extension from having a CFF table. I've also seen this in practice. Checking for the existence of the table is, IMHO, less error-prone.

On that note, I've also (weirdly) come across fonts with both `glyf` _and_ `CFF ` tables. No idea how that's supposed to work.

Reply:
You're right that file extension is used. It's an easy heuristic. Do you think we should change that to make it more sophisticated and closer to "technically correct"?

Reply:
I do. OTF and TTF aren't really different formats, they just contain different tables. Prawn would only have to read the directory structure at the beginning of the font to know if it contains a CFF table. Is that something you'd like to see in this PR?

Reply:
> I've also (weirdly) come across fonts with both glyf and CFF tables. No idea how that's supposed to work.

Maybe it's a hybrid font? You can treat it as TTF _or_ OTF and still get the same result? We should look into a few examples to verify if that's the case at least for the majority of cases. If that's so we should probably treat those as OTF fonts.

> Is that something you'd like to see in this PR?

Yes, please.

And maybe add a changelog entry?